massa-grpc/src/service.rs

```diff
@@ -0,0 +1,125 @@
+use crate::config::GrpcConfig;
+use crate::error::GrpcError;
+use futures_util::FutureExt;
+use massa_consensus_exports::{ConsensusChannels, ConsensusController};
+use massa_execution_exports::ExecutionController;
+use massa_pool_exports::{PoolChannels, PoolController};
+use massa_pos_exports::SelectorController;
+use massa_proto::massa::api::v1::grpc_server::GrpcServer;
+use massa_proto::massa::api::v1::FILE_DESCRIPTOR_SET;
+use massa_protocol_exports::ProtocolCommandSender;
+use massa_storage::Storage;
+use tokio::sync::oneshot;
+use tonic::codec::CompressionEncoding;
+use tonic_web::GrpcWebLayer;
+use tracing::log::{info, warn};
+
+/// gRPC API content
+pub struct MassaGrpcService {
+    /// link(channels) to the consensus component
+    pub consensus_controller: Box<dyn ConsensusController>,
+    /// link(channels) to the consensus component
+    pub consensus_channels: ConsensusChannels,
+    /// link to the execution component
+    pub execution_controller: Box<dyn ExecutionController>,
+    /// link(channels) to the pool component
+    pub pool_channels: PoolChannels,
+    /// link to the pool component
+    pub pool_command_sender: Box<dyn PoolController>,
+    /// link(channels) to the protocol component
+    pub protocol_command_sender: ProtocolCommandSender,
+    /// link to the selector component
+    pub selector_controller: Box<dyn SelectorController>,
+    /// link to the storage component
+    pub storage: Storage,
+    /// gRPC configuration
+    pub grpc_config: GrpcConfig,
+    /// node version
+    pub version: massa_models::version::Version,
+}
+
+impl MassaGrpcService {
+    /// Start the gRPC API
+    pub async fn serve(self, config: &GrpcConfig) -> Result<StopHandle, GrpcError> {
```

Comment:
IMHO, this function and stop handler should be in the to of the module. We'll implement a health service so this code will be incoherent

Reply:
@modship, what do you think